lib/drivers/onetouch/oneTouchUltraMini.js

```diff
@@ -585,13 +618,26 @@ module.exports = function (config) {
           }
 
           getRTC(function(err3, obj3) {
-            var currentDeviceTime = sundial.formatDeviceTime(new Date(obj3.timestamp * 1000));
-            common.checkDeviceTime(currentDeviceTime, cfg, function(err) {
-              progress(100);
-              data.getConfigInfo = true;
-              _.assign(data, obj2);
-              debug('getConfigInfo', data);
-              cb(err, data);
+            cfg.deviceInfo.deviceTime = sundial.formatDeviceTime(new Date(obj3.timestamp * 1000));
+            common.checkDeviceTime(cfg, function(err, serverTime) {
+              if (err === 'updateTime') {
+                var newDateTime = (serverTime.getTime() / 1000) + (sundial.getOffsetFromZone(serverTime, cfg.timezone) * 60);
+
+                setRTC(newDateTime, function(err4, obj4) {
+                  debug('Time set to', sundial.formatDeviceTime(new Date(obj4.timestamp * 1000)));
+                  progress(100);
```

Comment:
It would be nice (one day, sigh) to use event emitters here (which get listened for), instead of a callback :)

Reply:
Totally! :100: